x-pack/plugins/security/server/spaces/secure_spaces_client_wrapper.ts

```diff
@@ -265,6 +269,24 @@ export class SecureSpacesClientWrapper implements ISpacesClient {
       }
     }
 
+    // Fetch saved objects to be removed for audit logging
+    const finder = this.spacesClient.createSavedObjectFinder(id);
+    for await (const response of finder.find()) {
```

Comment:
Since this preflight check is potentially quite resource-intensive, can we add some logic to skip it if audit logging is not enabled?